crates/katana/executor/src/blockifier/utils.rs

```diff
@@ -62,7 +62,7 @@ pub fn estimate_fee(
     state: Box<dyn StateProvider>,
 ) -> Result<Vec<FeeEstimate>, TransactionExecutionError> {
     let state = CachedStateWrapper::new(StateRefDb::from(state));
-    let results = TransactionExecutor::new(&state, &block_context, false, transactions)
+    let results = TransactionExecutor::new(&state, &block_context, false, true, transactions)
```

Comment:
I think we should have the disable validation flag to affect fee estimation as well.

The usual flow when sending a tx is to send an estimate fee first, so if the user intentionally send a tx with invalid signature, even if they run Katana with validation off, they would still not pass the fee estimation as validation here is hardcoded to `true`. Which would definitely confuse the user.

Adding it here would also make sense because the v0.6.0 introduces a new [flag](https://github.com/starkware-libs/starknet-specs/blob/49665932a97f8fdef7ac5869755d2858c5e3a687/api/starknet_api_openrpc.json#L601) for the `estimateFee` endpoint.

Reply:
Addresses in 92362dfd27b7631aa763099d3b3e67d72cf47872.